File: src/main/java/com/devtoolcopilot/project/entity/GroupTask.java
package com.devtoolcopilot.project.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 群组任务（协作模块团队任务载体）。
 * 与个人任务体系(task)完全隔离，仅挂在群组维度，
 * 支持指派/认领/状态流转/截止时间。
 */
@Data
@TableName("group_task")
public class GroupTask {
    @TableId(type = IdType.AUTO)
    private Long id;

    private Long groupId;

    private String title;

    private String description;

    /** HIGH / MEDIUM / LOW */
    private String priority;

    /** TODO / DOING / DONE */
    private String status;

    /** 指派人（可空，空表示可被成员认领） */
    private Long assigneeId;

    private Long createdBy;

    private LocalDateTime dueTime;

    private LocalDateTime doneTime;

    private Long sort;

    @TableField("create_time")
    private LocalDateTime createTime;

    @TableField("update_time")
    private LocalDateTime updateTime;
}

File: devtool-copilot-web/src/api/groupTask.ts
import { apiDelete, apiGet, apiPost, apiPut } from './http'

export type GroupTaskStatus = 'TODO' | 'DOING' | 'DONE'
export type GroupTaskPriority = 'HIGH' | 'MEDIUM' | 'LOW'

export interface GroupTaskItem {
  id: number
  groupId: number
  title: string
  description?: string | null
  priority: GroupTaskPriority
  status: GroupTaskStatus
  assigneeId?: number | null
  assigneeUsername?: string | null
  createdBy?: number | null
  creatorUsername?: string | null
  dueTime?: string | null
  doneTime?: string | null
  overdue: boolean
  createTime?: string | null
  updateTime?: string | null
}

export interface GroupTaskBoardResponse {
  todo: GroupTaskItem[]
  doing: GroupTaskItem[]
  done: GroupTaskItem[]
  total: number
  doingCount: number
  doneCount: number
  overdueCount: number
  completionRate: number
}

export interface GroupTaskCreatePayload {
  title: string
  description?: string
  priority?: GroupTaskPriority
  assigneeId?: number | null
  dueTime?: string | null
}

export interface GroupTaskUpdatePayload {
  title?: string
  description?: string
  priority?: GroupTaskPriority
  dueTime?: string | null
}

export const groupTaskApi = {
  board(groupId: number) {
    return apiGet<GroupTaskBoardResponse>(`/api/project/team/groups/${groupId}/tasks`)
  },
  create(groupId: number, payload: GroupTaskCreatePayload) {
    return apiPost<GroupTaskItem>(`/api/project/team/groups/${groupId}/tasks`, payload)
  },
  update(groupId: number, taskId: number, payload: GroupTaskUpdatePayload) {
    return apiPut<void>(`/api/project/team/groups/${groupId}/tasks/${taskId}`, payload)
  },
  assign(groupId: number, taskId: number, assigneeId: number) {
    return apiPost<void>(`/api/project/team/groups/${groupId}/tasks/${taskId}/assign`, { assigneeId })
  },
  claim(groupId: number, taskId: number) {
    return apiPost<void>(`/api/project/team/groups/${groupId}/tasks/${taskId}/claim`)
  },
  changeStatus(groupId: number, taskId: number, status: GroupTaskStatus) {
    return apiPost<void>(`/api/project/team/groups/${groupId}/tasks/${taskId}/status`, { status })
  },
  delete(groupId: number, taskId: number) {
    return apiDelete<void>(`/api/project/team/groups/${groupId}/tasks/${taskId}`)
  }
}
